core/src/main/java/com/facebook/testing/screenshot/internal/DefaultDeviceIdentifier.java

```diff
@@ -0,0 +1,47 @@
+package com.facebook.testing.screenshot.internal;
+
+import android.content.Context;
+import android.content.pm.PackageInfo;
+import android.content.pm.PackageManager;
+import android.os.Build;
+import android.util.DisplayMetrics;
+import android.view.WindowManager;
+
+import com.facebook.testing.screenshot.DeviceIdentifier;
+
+public class DefaultDeviceIdentifier implements DeviceIdentifier {
+
+  private static final String PLAY_SERVICES = "GP";
+  private static final String NO_PLAY_SERVICES = "NO_GP";
+
+  private final Context context;
+
+  DefaultDeviceIdentifier(Context context) {
+    this.context = context;
+  }
+
+  @Override
+  public String generateId() {
+    final int sdkVersion = Build.VERSION.SDK_INT;
+    final String googlePlayServicesString = obtainPlayServicesString();
+    final String metricsString = obtainMetricsString();
+    final String sdkVersionString = String.valueOf(sdkVersion);
+    return String.format("API_%s_%s_%s", sdkVersionString, googlePlayServicesString, metricsString);
```

Comment:
this isn't reliable enough as an identifier.. it might seem like it's better than no emulator identifier, but users might get confused and assume that just these three pieces of information guarrantee that the screenshots will be the same. It's not. (In particular, this also misses out on screen density which is a huge deal)

Reply:
In my new PR I have added screen density and I have kept the ones in this DefaultDeviceIdentifier class.
I will explain in the new PR how extract those params from adb